lib/ratelimit.c

```diff
@@ -63,3 +63,50 @@ tb_ratelimit_allow(struct token_bucket_ratelimit_state *tbrs)
 
 	return rc;
 }
+
+void
+log_ratelimit_state_init(struct log_ratelimit_state *lrs,
+	uint32_t ratelimit_level, uint32_t interval, uint32_t burst)
+{
+	lrs->ratelimit_level = ratelimit_level;
+	lrs->interval = interval;
+	lrs->burst = burst;
+	lrs->printed = 0;
+	lrs->missed = 0;
+	lrs->begin = 0;
+}
+
+/*
+ * Rate limiting log entries.
+ *
+ * Returns:
+ * - true means go ahead and do it.
+ * - false means callbacks will be suppressed.
+ */
+bool
+log_ratelimit_allow(struct log_ratelimit_state *lrs)
+{
+	uint64_t now;
+
+	if (!lrs->interval)
+		return true;
+
+	now = rte_rdtsc();
+
+	if (!lrs->begin)
+		lrs->begin = now;
+
+	if (lrs->begin + lrs->interval * cycles_per_ms < now) {
```

Comment:
Why don't we rename `interval` to `interval_cycles` and initialize it with `lrs->interval * cycles_per_ms` to avoid this multiplication every call since `interval` is only really used here?